docs/settings.md

```diff
@@ -540,15 +540,17 @@ are of the form `'{placeholdername}'`. The following placeholders can be used:
 Controls if we try to retrieve built-in flags from a target compiler. If a
 compiler is not `null`, we try to ask it for the defines and include paths it
 sets implicitly and pass them to the clang compiler which is used to generate
-the code completion. If your completions require the knowledge about the
+the code completion. If you need to pass additional flags to the target 
+compiler, you can do so by providing a list of flags.
+If your completions require the knowledge about the
 toolchain, this option should improve the quality of the completions, however,
 in some corner cases it might cause completions to fail entirely.
 
 !!! example "Default value"
     ```json
     "target_compilers": {
-        "C":              null,
-        "CPP":            null,
+        "C":              "clang",
+        "CPP":            {"compiler": "clang", "flags": ["-flag1", "-flag2"]},
```

Comment:
This default value is different from the one above in the actual settings. I'd suggest to add a different example with all the actual flags filled in (just follow your usecase and fill the actual compiler and flags that work for you) as as separate code snippet, leaving the "Default value" snippet being equal to the one already in the settings.